src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -499,7 +499,11 @@ export const duplicateFormField = (
     },
   ).andThen((updatedForm) => {
     if (!updatedForm) {
-      return errAsync(new FormNotFoundError())
+      // Null return could be due to form or field not existing.
+      // Hence, check if field exists in form and return the appropriate error
+      return getFormField(form, fieldId).asyncAndThen(() =>
```

Comment:
i think better comments might make this clearer. 

```ts
return getFormField(form, fieldId)
  // Success means field is in initial form object; but query still returned null.
  // Return best guess error that form is now not found in the DB.
  .asyncAndThen(() => errAsync(new FormNotFoundError()))
  // Otherwise, err(FieldNotFoundError) will be returned by the function.
```

Reply:
fixed, thanks!